core/unit_test/TestGraph.hpp

```diff
@@ -867,6 +903,160 @@ TEST(TEST_CATEGORY, when_all_type) {
   static_assert(std::is_same_v<decltype(tail), node_ref_tail_t>);
 }
 
+#if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP)
+template <GraphNodeType value, typename DstType, typename... SrcTypes>
+__global__ void set_to(DstType* const dst, const SrcTypes* const... srcs) {
+  dst[threadIdx.y] += (srcs[threadIdx.y] + ...) + static_cast<DstType>(value);
+}
+#endif
+
+template <typename Exec>
+struct ExternalCapture {
+#if defined(KOKKOS_ENABLE_CUDA) ||                                           \
+    (defined(KOKKOS_ENABLE_HIP) && defined(KOKKOS_IMPL_HIP_NATIVE_GRAPH)) || \
+    (defined(KOKKOS_ENABLE_SYCL) && defined(SYCL_EXT_ONEAPI_GRAPH))
+  // clang-format off
+  template <typename Pred, typename DstType, typename... SrcTypes>
+  static auto add(const Pred& pred, const Exec& exec, const DstType& dst, SrcTypes&&... srcs) {
+    return pred.
+#if defined(KOKKOS_ENABLE_CUDA)
+    cuda_capture
+#elif defined(KOKKOS_ENABLE_HIP)
+    hip_capture
+#elif defined(KOKKOS_ENABLE_SYCL)
+    sycl_capture
+#endif
+    (exec, [dst, tup = std::make_tuple(std::forward<SrcTypes>(srcs)...)](const Exec& exec_) {   \
+            std::apply([&](const auto&... args) {                                                         \
+                ExternalCapture::compute(exec_, dst.data(), args.data()...);}, tup);                      \
+    });                                                                                                   \
```

Comment:
The alignment of the `\` seems a bit off?

Reply:
Indeed, and done :heavy_check_mark: